corecito.py

```diff
@@ -4,88 +4,97 @@
 
 
       ###########################
       # Core Number Adjustments #
       ###########################
-      deviated_core_number = base_currency_available / buy_price
+      deviated_core_number = balances['base_currency_available'] / buy_price
       logger.info(f'Core number adjustments')
-      logger.info(f'Core number: {core_number} {core_number_currency}')
-      logger.info(f'Deviated Core number:{deviated_core_number:.6f} {core_number_currency}')
-      excess = round(deviated_core_number - core_number, config['max_decimals_buy'])
-      increase_percentage = excess * 100 / core_number
-      missing = round(core_number - deviated_core_number, config['max_decimals_sell'])
-      decrease_percentage = missing * 100 / core_number
-
-      if coreNumberExploded(core_number, deviated_core_number, config['max_core_number_increase_percentage']):
+      logger.info(f'Core number: {account.core_number} {account.core_number_currency}')
+      logger.info(f'Deviated Core number:{deviated_core_number:.6f} {account.core_number_currency}')
+      excess = round(deviated_core_number - account.core_number, account.max_decimals_buy)
+      increase_percentage = excess * 100 / account.core_number
+      missing = round(account.core_number - deviated_core_number, account.max_decimals_sell)
+      decrease_percentage = missing * 100 / account.core_number
+
+      if coreNumberExploded(account.core_number, deviated_core_number, account.max_core_number_increase_percentage):
         logger.info(f'> Exploded {increase_percentage:.2f}%\nConsider updating CoreNumber to {deviated_core_number:.6f}')
 
-      elif coreNumberIncreased(core_number, deviated_core_number, config['min_core_number_increase_percentage'], config['max_core_number_increase_percentage']):
-        logger.info(f'Increased {increase_percentage:.2f}% - excess of {excess:.6f} {core_number_currency} denominated in {base_currency}')
-        tx_result = round(excess * buy_price, config['max_decimals_buy'])
-        logger.info(f'\n\n>>> Selling: {tx_result:.6f} {base_currency} at {buy_price} to park an excess of {excess:.6f} {core_number_currency}\n')
+      elif coreNumberIncreased(account.core_number, deviated_core_number, account.min_core_number_increase_percentage, account.max_core_number_increase_percentage):
+        logger.info(f'Increased {increase_percentage:.2f}% - excess of {excess:.6f} {account.core_number_currency} denominated in {account.base_currency}')
+        tx_result = round(excess * buy_price, account.max_decimals_buy)
+        logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at {buy_price} to park an excess of {excess:.6f} {account.core_number_currency}\n')
         # Sell excess of base currency ie. => in ETH_BTC pair, sell excess BTC => Buy ETH
         if (not config['safe_mode_on']):
-          account.order_market_buy(symbol=pair, quantity=excess)
+          await account.order_market_buy(quantity=excess)
```

Comment:
The quantity here is tricky, in Binance it's the excess (as it should be) but for some weird reason (API bug?) in crypto.com, the quantity has to be the `tx_result`

Reply:
ooops, great that you caught it. I've made a fix for that. What do you think?